src/PrestaShopBundle/Routing/ModuleLoader.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+namespace PrestaShopBundle\Routing;
+
+use Symfony\Component\Config\Loader\Loader;
+use Symfony\Component\Routing\RouteCollection;
+
+/**
+ * This class is responsible of loading routes of enabled modules.
+ */
+class ModuleLoader extends Loader
+{
+    /**
+     * @var array $activeModules The list of activated modules.
+     */
+    private $activeModulesPaths;
+
+    /**
+     * @var bool $isLoaded We load the route collection only once per request.
+     */
+    private $isLoaded = false;
+
+    /**
+     * @var string $modulePath The path to the modules folder.
+     */
+    private $modulePath;
+
+    public function __construct(array $activeModulesPaths, $modulePath)
+    {
+        $this->activeModulesPaths = $activeModulesPaths;
```

Comment:
i think we should be able to access installed but disabled module's controllers as well, since they can be configured. what do you think?

Reply:
Nah, we shouldn't load them in my opinion, because the configuration page of a module is basically not in a controller.